hivemq-edge/src/main/java/com/hivemq/configuration/reader/ConfigFileReaderWriter.java

```diff
@@ -119,8 +127,32 @@ public ConfigFileReaderWriter(
         this.internalConfigurator = internalConfigurator;
     }
 
-    public @NotNull HiveMQConfigEntity applyConfig() {
-        return readConfigFromXML();
+    public HiveMQConfigEntity applyConfig() {
+        final HiveMQConfigEntity hiveMQConfigEntity = readConfigFromXML();
+        this.configEntity = hiveMQConfigEntity;
+        setConfiguration(hiveMQConfigEntity);
+        return hiveMQConfigEntity;
+    }
+
+    public void applyConfigAndWatch(final long checkIntervalInMs) {
+        if(scheduledExecutorService != null) {
+            throw new IllegalStateException("Config watch was already started");
+        }
+        final long interval = 0 >= checkIntervalInMs ? checkIntervalInMs : 0;
```

Comment:
Is 0 a sane value here?

Reply:
It essentially means "as fast as possible". Since I am using a single threaded pool it means that it will be invoked as often as possible.
We will see if that makes sense.